src/lazy/bytes_ref.rs

```diff
@@ -0,0 +1,123 @@
+use crate::text::text_formatter::IonValueFormatter;
+use crate::Bytes;
+use std::borrow::Cow;
+use std::fmt::{Debug, Display, Formatter};
+use std::ops::Deref;
+
+pub struct BytesRef<'data> {
+    data: Cow<'data, [u8]>,
+}
```

Comment:
🗺️ When there was only a binary reader, methods that read blobs could always return a `&[u8]`--a slice of the input buffer. Now that we also have a text reader, we need to accommodate base64-encoded blobs, which always require a new `Vec` to be allocated to hold the decoded data. `BytesRef` can hold either a borrowed `&[u8]` or an owned `Vec<u8>`, allowing it to be used in either situation.

This type is analogous to `StrRef` and `SymbolRef` but for blobs.